python/src/aiconfig/editor/client/src/components/AIConfigEditor.tsx

```diff
@@ -828,6 +828,57 @@ export default function EditorContainer({
     return () => clearInterval(interval);
   }, [getServerStatusCallback, serverStatus]);
 
+  const runningPromptId: string | undefined = aiconfigState._ui.runningPromptId;
+  const promptListContainer = (
+    <Container maw="80rem" className={classes.promptsContainer}>
+      <div className={classes.addPromptRow}>
+        <AddPromptButton
+          getModels={callbacks.getModels}
+          addPrompt={(model: string) => onAddPrompt(0, model)}
+        />
+      </div>
+      {aiconfigState.prompts.map((prompt: ClientPrompt, i: number) => {
+        return (
+          <Stack key={prompt._ui.id}>
+            <Flex mt="md">
+              <PromptMenuButton
+                promptId={prompt._ui.id}
+                onDeletePrompt={() => onDeletePrompt(prompt._ui.id)}
+              />
+              <PromptContainer
+                prompt={prompt}
+                getModels={callbacks.getModels}
+                onChangePromptInput={onChangePromptInput}
+                onChangePromptName={onChangePromptName}
+                cancel={callbacks.cancel}
+                onRunPrompt={onRunPrompt}
+                onUpdateModel={onUpdatePromptModel}
+                onUpdateModelSettings={onUpdatePromptModelSettings}
+                onUpdateParameters={onUpdatePromptParameters}
+                defaultConfigModelName={aiconfigState.metadata.default_model}
+                isRunButtonDisabled={
+                  runningPromptId !== undefined &&
+                  runningPromptId !== prompt._ui.id
```

Comment:
cc @rholinshead I feel confident I don't need to do the explicit check for `runningPromptId !== undefined`, but was just worried in case we can have situations where any of the `prompt._ui.id` fields could be undefined. Is that possible?

Reply:
No need for the check -- _ui.id is required to exist in ClientPrompt